functests/xt_u32/xt_u32.go

```diff
@@ -0,0 +1,223 @@
+			Expect(err).ToNot(HaveOccurred())
+		}
+	})
+
+	BeforeEach(func() {
+		if !hasNonCnfWorkers {
+			Skip("Skipping as no non-enabled nodes are available")
+		}
+
+		namespaces.Clean(TestNamespace, "xt-u32", client.Client)
+		By("Choosing the test node")
+		nodes, err := client.Client.Nodes().List(context.Background(), metav1.ListOptions{
+			LabelSelector: xt_u32NodeSelector,
+		})
+		Expect(err).ToNot(HaveOccurred())
+
+		filtered, err := utilNodes.MatchingOptionalSelector(nodes.Items)
+		Expect(err).ToNot(HaveOccurred())
+
+		if discovery.Enabled() && len(filtered) == 0 {
+			Skip("Did not find a node without xt_u32 module enabled")
+		} else {
+			Expect(len(filtered)).To(BeNumerically(">", 0))
+		}
+		testNode = filtered[0].ObjectMeta.Labels[hostnameLabel]
+	})
+
+	Context("Validate the module is enabled and works", func() {
+		It("Should create an iptables rule inside a pod that has the module enabled", func() {
+			By("Create a pod")
+			pod := xt_u32TestPod("xt-u32", testNode, TestNamespace, []string{"sleep inf"})
+			xt_u32Pod, err := client.Client.Pods(TestNamespace).Create(context.Background(), pod, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Check if the pod is running")
+			Eventually(func() k8sv1.PodPhase {
+				res, err := client.Client.Pods(TestNamespace).Get(context.Background(), xt_u32Pod.Name, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return res.Status.Phase
+			}, 3*time.Minute, 1*time.Second).Should(Equal(k8sv1.PodRunning))
+
+			By("Check that xt_u32 module is loaded")
+			cmd := []string{"lsmod"}
+			out, err := pods.ExecCommand(client.Client, *xt_u32Pod, cmd)
+			Expect(err).ToNot(HaveOccurred())
+			res := strings.Contains(out.String(), "xt_u32")
+			if !res {
+				err = errors.New("xt_u32 module is not loaded")
+				Expect(err).ToNot(HaveOccurred())
+			}
+
+			By("Create an iptables rule within the pod")
+			cmd = []string{"iptables", "-t", "filter", "-A", "INPUT", "-i", "eth0", "-m",
+				"u32", "--u32", "6 & 0xFF = 1 && 4 & 0x3FFF = 0 && 0 >> 22 & 0x3C @ 0 >> 24 = 8",
+				"-j", "DROP"}
+			_, err = pods.ExecCommand(client.Client, *xt_u32Pod, cmd)
+			Expect(err).ToNot(HaveOccurred())
+		})
+	})
+})
+
+func findXT_U32NodeSelector() (string, error) {
+	mcList, err := client.Client.MachineConfigs().List(context.Background(), metav1.ListOptions{})
+	Expect(err).ToNot(HaveOccurred())
+
+	for _, mc := range mcList.Items {
+		enables, err := enablesXT_U32(mc)
+		if err != nil {
+			return "", err
+		}
+		if enables {
+			mcLabel, found := mc.ObjectMeta.Labels["machineconfiguration.openshift.io/role"]
+			if !found {
+				continue
+			}
+
+			xt_u32NodeSelector, err := findXT_U32NodeSelectorByMCLabel(mcLabel)
+			if err != nil {
+				continue
+			}
+
+			return xt_u32NodeSelector, nil
+		}
+	}
+
+	return "", errors.New("Cannot find a machine configuration that enables XT_U32")
+}
+
+func enablesXT_U32(mc mcfgv1.MachineConfig) (bool, error) {
```

Comment:
IIUC this function only try to see if it's configure or not, returning true/false, based on that
so the name is confusing
is_xt_u32_enabled is probably better

Reply:
But with camelCase, please :-)